x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/Connector.java

```diff
@@ -0,0 +1,477 @@
+     * @param apiKeyId           API key ID used for authentication/authorization against ES.
+     * @param configuration      Configuration settings for the connector.
+     * @param customScheduling   Custom scheduling settings for the connector.
+     * @param description        Description of the connector.
+     * @param error              Information about the last error encountered by the connector, if any.
+     * @param features           Features enabled for the connector.
+     * @param filtering          Filtering settings applied by the connector.
+     * @param indexName          Name of the index associated with the connector.
+     * @param isNative           Flag indicating whether the connector is a native type.
+     * @param language           The language supported by the connector.
+     * @param syncInfo           Information about the synchronization state of the connector.
+     * @param name               Name of the connector.
+     * @param pipeline           Ingest pipeline configuration.
+     * @param scheduling         Scheduling settings for regular data synchronization.
+     * @param serviceType        Type of service the connector integrates with.
+     * @param status             Current status of the connector.
+     * @param syncCursor         Position or state indicating the current point of synchronization.
+     * @param syncNow            Flag indicating whether an immediate synchronization is requested.
+     */
+    private Connector(
+        String connectorId,
+        String apiKeyId,
+        Map<String, Object> configuration,
+        ConnectorCustomSchedule customScheduling,
+        String description,
+        String error,
+        ConnectorFeatures features,
+        List<ConnectorFiltering> filtering,
+        String indexName,
+        boolean isNative,
+        String language,
+        ConnectorSyncInfo syncInfo,
+        String name,
+        ConnectorIngestPipeline pipeline,
+        ConnectorScheduling scheduling,
+        String serviceType,
+        ConnectorStatus status,
+        Object syncCursor,
+        boolean syncNow
+    ) {
+        this.connectorId = Objects.requireNonNull(connectorId, "connectorId cannot be null");
+        ;
+        this.apiKeyId = apiKeyId;
+        this.configuration = Objects.requireNonNullElse(configuration, Collections.emptyMap());
+        this.customScheduling = customScheduling;
+        this.description = description;
+        this.error = error;
+        this.features = features;
+        this.filtering = filtering;
+        this.indexName = indexName;
+        this.isNative = isNative;
+        this.language = language;
+        this.syncInfo = Objects.requireNonNullElse(syncInfo, new ConnectorSyncInfo.Builder().build());
+        this.name = name;
+        this.pipeline = pipeline;
+        this.scheduling = scheduling;
+        this.serviceType = serviceType;
+        this.status = status;
```

Comment:
Do we want to validate `status` is non-null here, or define a default? (`status` is not annotated as `@Nullable`)